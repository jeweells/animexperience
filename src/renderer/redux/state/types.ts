import { FStatus, Optional, TopView } from '@shared/types'
import {
  AnimeInfo,
  DeepAnimeIdSearchResult,
  EpisodeInfo,
  FollowedAnime,
  RecentAnimeInfo,
  RecommendationInfo,
  WatchHistoryItem
} from '@shared/types'
import { FollowedAnimeWStatus } from '../reducers/followedAnimes'
import { PreAllowWatch } from '../reducers/invokedLink'
import { RecentAnimeData } from '../../src/hooks/useRecentAnimes'
import { VideoOption } from '../../src/components/VideoPlayer'

export type AnimeSearchState = {
  // Boolean id
  searching?: string
  result?: Optional<DeepAnimeIdSearchResult>
  status: Partial<{
    result: FStatus
    moreResults: FStatus
  }>
}

export type FollowedAnimesState = {
  followedDict: Record<string, FollowedAnime>
  followed: FollowedAnimeWStatus[]
  status: Partial<{
    followed: FStatus
  }>
}

export type InvokedLinkState = {
  open: {
    watch?: string
  }
  preAllow: Partial<{
    watch: PreAllowWatch
  }>
  status: {
    watch?: FStatus
  }
}

export type NotificationsState = {
  message: string | null
  key: string
}

export type PeekState = {
  name?: string
  status: Partial<{
    info: FStatus
  }>
  info?: Optional<AnimeInfo>
  peeking?: string
}

export type PlayerState = {
  open?: string
  freezed?: boolean
}

export type OptionInfo = {
  name: string
  prefer?: boolean
  score: number
}
// Define a type for the slice state
export type PlayerOptionsState = {
  options?: OptionInfo[]
  history?: string[]
  preferred?: string[]
  status: Partial<{
    options: FStatus
  }>
}

export type RecommendationsState = {
  [name: string]: Partial<{
    status: FStatus
    recommendations: RecommendationInfo[]
  }>
}

export type TopviewState = {
  views: TopView[]
}

export type WatchState = {
  watching?: Optional<RecentAnimeData>
  availableVideos?: Optional<VideoOption[]>
  status: {
    availableVideos?: FStatus
    info?: FStatus
    watchEpisode?: FStatus
  }
  info?: Optional<AnimeInfo>
  showNextEpisodeButton?: boolean
  nextEpisodeTimeout: number
  autoFullScreen?: boolean
}

export type WatchedState = {
  episodes: {
    [name: string]: EpisodeInfo
  }
  recently?: RecentAnimeInfo[]
  status: {
    episodes?: FStatus
    recently?: FStatus
  }
}

export type WatchHistoryState = {
  sorted?: WatchHistoryItem[]
  status: {
    sorted?: FStatus
  }
}
