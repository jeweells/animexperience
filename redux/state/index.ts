import {
    AnimeSearchState,
    FollowedAnimesState,
    InvokedLinkState,
    NotificationsState,
    PeekState,
    PlayerOptionsState,
    PlayerState,
    RecommendationsState,
    TopviewState,
    WatchedState,
    WatchHistoryState,
    WatchState,
} from './types'

const animeSearch: AnimeSearchState = {
    status: {},
}

const followedAnimes: FollowedAnimesState = {
    followed: [],
    followedDict: {},
    status: {},
}

const invokedLink: InvokedLinkState = {
    open: {},
    preAllow: {},
    status: {},
}

const notifications: NotificationsState = {
    message: null,
    key: '_',
}

const peek: PeekState = {
    status: {},
}

const player: PlayerState = {}

const playerOptions: PlayerOptionsState = {
    status: {},
}

const recommendations: RecommendationsState = {}

const topView: TopviewState = {
    views: [],
}

const watch: WatchState = {
    status: {},
    nextEpisodeTimeout: -1,
}

const watched: WatchedState = {
    episodes: {},
    status: {},
}

const watchHistory: WatchHistoryState = {
    status: {},
}

export const initialState = {
    animeSearch,
    followedAnimes,
    invokedLink,
    notifications,
    peek,
    player,
    playerOptions,
    recommendations,
    topView,
    watch,
    watched,
    watchHistory,
}

export type RootState = typeof initialState
