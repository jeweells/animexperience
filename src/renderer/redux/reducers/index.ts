export { animeSearch } from './animeSearch'
export { followedAnimes } from './followedAnimes'
export { invokedLink } from './invokedLink'
export { notifications } from './notifications'
export { peek } from './peek'
export { player } from './player'
export { playerOptions } from './playerOptions'
export { recommendations } from './recommendations'
export { topView } from './topView'
export { watch } from './watch'
export { watched } from './watched'
export { watchHistory } from './watchHistory'
