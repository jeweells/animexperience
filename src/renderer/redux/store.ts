import { configureStore } from '@reduxjs/toolkit'
import animeSearch from './reducers/animeSearch'
import followedAnimes from './reducers/followedAnimes'
import peek from './reducers/peek'
import player from './reducers/player'
import playerOptions from './reducers/playerOptions'
import recommendations from './reducers/recommendations'
import topView from './reducers/topView'
import watch from './reducers/watch'
import watched from './reducers/watched'
import watchHistory from './reducers/watchHistory'
import invokedLink from './reducers/invokedLink'
import notifications from './reducers/notifications'

export default configureStore({
  reducer: {
    watch,
    player,
    watched,
    recommendations,
    watchHistory,
    peek,
    topView,
    playerOptions,
    animeSearch,
    followedAnimes,
    invokedLink,
    notifications
  }
})
