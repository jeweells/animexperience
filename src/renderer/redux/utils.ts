import { ThunkDispatch } from 'redux-thunk'
import { RootState } from './state'
import { AnyAction } from '@reduxjs/toolkit'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { ForcedAny } from '@shared/types'

export type AppDispatch = ThunkDispatch<RootState, ForcedAny, AnyAction>

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export const useAppDispatch = useDispatch as () => AppDispatch
