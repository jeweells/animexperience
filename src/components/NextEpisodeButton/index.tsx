import React from 'react'
import { watch } from '@reducers'
import { useAppDispatch, useAppSelector } from '~/redux/utils'
import font from '../../fonts/Quicksand/Quicksand-Light.ttf'
import Fade from '@mui/material/Fade'

let currUrl = ''
const updateUrl = () => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    currUrl = require('@electron/remote').getCurrentWindow().webContents.getURL()
}
// We need to form an absolute url since this element will be injected inside an iframe
// and inside iframes, relatives url won't match ours
export const quicksandCss = () => `
    @font-face {
        font-family: 'Quicksand';
        src: url('${new URL(font, currUrl).href}') format('truetype');
        font-weight: 300;
        font-style: normal;
    }
`

export const buttonCss = `
   
    button.raex-button-injected {
        font-family: 'Quicksand';
        color: #e9ebf0;
        background: #292d33;
        display: inline-block;
        margin-bottom: 0;
        font-weight: normal;
        text-align: center;
        vertical-align: middle;
        cursor: pointer;
        outline: 0 !important;
        white-space: nowrap;
        border: none;
        user-select: none;
        padding: 8px 12px;
        font-size: 14px;
        line-height: 1.42857143;
        border-radius: 6px;
        position: relative;
        transition: color 0.2s linear, background-color 0.3s linear;
        margin: 0;
        overflow: hidden;
        text-decoration: none;
    }
    button.raex-button-injected:hover, button.raex-button-injected:focus {
        color: #e9ebf0;
        background-color: #3c3f43;
        
    }
`

export type NextEpisodeButtonProps = {}

export const NextEpisodeButton: React.FC<NextEpisodeButtonProps> = React.memo(({}) => {
    const timeout = useAppSelector((d) => d.watch.nextEpisodeTimeout)
    const max = useAppSelector((d) => d.watch.info?.episodesRange?.max) ?? 0
    const episode = useAppSelector((d) => d.watch.watching?.episode) ?? 0
    const dispatch = useAppDispatch()
    const showNextButton =
        useAppSelector((d) => d.watch.showNextEpisodeButton) && episode < max
    const handleNext = () => {
        dispatch(watch.setNextEpisodeButton(false))
        dispatch(watch.nextEpisode())
    }
    React.useMemo(updateUrl, [])

    return (
        <Fade in={showNextButton} timeout={400}>
            <div
                style={{
                    position: 'absolute',
                    top: 16,
                    right: 16,
                    zIndex: 2147483647,
                }}
            >
                <style>
                    {quicksandCss()}
                    {buttonCss}
                </style>
                <button
                    className={'raex-button-injected'}
                    onClick={() => {
                        handleNext()
                    }}
                >
                    <span style={{ position: 'relative', zIndex: 2 }}>
                        Siguiente episodio
                    </span>
                    <div
                        style={{
                            position: 'absolute',
                            zIndex: 1,
                            top: 0,
                            left: 0,
                            bottom: 0,
                            backgroundColor: 'rgba(255,255,255,0.1)',
                            width: timeout !== -1 && showNextButton ? '100%' : '0',
                            transition: timeout === -1 ? 'none' : `all ${timeout}s`,
                        }}
                    />
                </button>
            </div>
        </Fade>
    )
})

NextEpisodeButton.displayName = 'NextEpisodeButton'

export default NextEpisodeButton
