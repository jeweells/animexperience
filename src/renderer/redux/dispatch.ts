import { AppDispatch } from '~/redux/utils'
import store from '~/redux/store'

export const dispatch: AppDispatch = (action) => store.dispatch(action)
