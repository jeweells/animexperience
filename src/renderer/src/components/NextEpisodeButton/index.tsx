import * as React from 'react'
import { watch } from '@reducers'
import { useAppDispatch, useAppSelector } from '~/redux/utils'
import Fade from '@mui/material/Fade'
import { NEXT_EPISODE_BUTTON } from '@selectors'
import { useEffect, useState } from 'react'
import Button from '@mui/material/Button'
import {
  usePlayPause,
  useSeek,
  useShowControls
} from '@components/VideoPlayer/components/Controls/hooks'
import { canGoToNextEpisode } from '@components/VideoPlayer/hooks'
import { SECONDS_NEXT_BUTTON_DISPLAY } from '@components/VideoPlayer/constants'

export const NextEpisodeButton: React.FC = React.memo(() => {
  const [{ seconds, clear }, _setTimeout] = useState<{ seconds: number; clear?: () => void }>({
    seconds: -1
  })
  const max = useAppSelector((d) => d.watch.info?.episodesRange?.max) ?? 0
  const episode = useAppSelector((d) => d.watch.watching?.episode) ?? 0
  const dispatch = useAppDispatch()
  const handleNext = () => {
    dispatch(watch.nextEpisode())
  }

  const { isPlaying } = usePlayPause()
  const { show, transitionMs } = useShowControls()
  const { time } = useSeek()
  const showNextButton = episode < max && time.duration > 0 && canGoToNextEpisode(time)

  useEffect(() => {
    if (!showNextButton) return
    const t = setTimeout(handleNext, SECONDS_NEXT_BUTTON_DISPLAY * 1000)
    _setTimeout({
      seconds: SECONDS_NEXT_BUTTON_DISPLAY,
      clear: () => {
        clearTimeout(t)
        _setTimeout({ seconds: -1 })
      }
    })
    return () => {
      clearTimeout(t)
    }
  }, [showNextButton])

  useEffect(() => {
    const handleMouseMove = () => {
      clear?.()
    }
    document.addEventListener('mousemove', handleMouseMove)
    return () => {
      document.removeEventListener('mousemove', handleMouseMove)
    }
  }, [clear])

  useEffect(() => {
    if (isPlaying) return
    clear?.()
  }, [isPlaying, clear])

  return (
    <Fade in={showNextButton} timeout={400}>
      <div>
        <div
          style={{
            position: 'absolute',
            right: 16,
            bottom: 16,
            transform: show ? 'translateY(-56px)' : 'translateY(0)',
            transition: `all ${transitionMs}ms ease`
          }}
        >
          <Button
            style={{ filter: 'invert(100%)', overflow: 'hidden' }}
            data-testid={NEXT_EPISODE_BUTTON.BUTTON}
            onClick={() => {
              handleNext()
            }}
          >
            <span style={{ position: 'relative', zIndex: 2 }}>Siguiente episodio</span>
            <div
              style={{
                position: 'absolute',
                zIndex: 1,
                top: 0,
                left: 0,
                bottom: 0,
                backgroundColor: 'rgba(0,0,0,0.9)',
                width: seconds !== -1 && showNextButton ? '100%' : '0',
                transition: seconds === -1 ? 'none' : `all ${seconds}s`
              }}
            />
          </Button>
        </div>
      </div>
    </Fade>
  )
})

NextEpisodeButton.displayName = 'NextEpisodeButton'

export default NextEpisodeButton
