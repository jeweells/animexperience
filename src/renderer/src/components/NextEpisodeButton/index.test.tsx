import NextEpisodeButton from './index'
import { Provider } from 'react-redux'
import { RootState } from '~/redux/state'
import { DeepPartial } from 'redux'
import { MockStoreEnhanced } from 'redux-mock-store'
import { NEXT_EPISODE_BUTTON } from '@selectors'

describe('NextEpisodeButton', () => {
  let initialState: DeepPartial<RootState>
  let store: MockStoreEnhanced

  beforeEach(() => {
    initialState = {
      watch: {
        showNextEpisodeButton: true,
        nextEpisodeTimeout: 5,
        info: {
          description: 'here goes description',
          link: 'here goes link',
          emitted: {
            from: 69,
            to: 129
          },
          related: [
            { name: 'Related name', link: 'related link', type: 'Ova' },
            { name: 'Related name', link: 'related link', type: 'Serie' },
            { name: 'Related name', link: 'related link', type: 'Película' },
            { name: 'Related name', link: 'related link', type: 'Especial' }
          ],
          episodeLink: 'here goes episode link (here goes episode replace)',
          type: 'Serie',
          image: 'here goes image',
          episodeReplace: 'here goes episode replace',
          episodesRange: {
            max: 10,
            min: 1
          },
          otherTitles: ['title1', 'title2'],
          status: 'En emisión',
          tags: ['tag1', 'tag2'],
          thumbnail: 'here goes thumbnail',
          title: 'here goes title'
        },
        watching: {
          date: '01/01/2022',
          episode: 5,
          img: 'image link',
          link: 'anime link',
          name: 'anime name'
        }
      }
    }
  })

  const getComponent = () => {
    store = mockStore(initialState)
    return (
      <Provider store={store}>
        <NextEpisodeButton />
      </Provider>
    )
  }

  it('renders default', async () => {
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders default when not shown', async () => {
    initialState.watch!.showNextEpisodeButton = false
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders when the episode is the last one', async () => {
    initialState.watch!.info!.episodesRange!.max = initialState.watch!.watching!.episode!
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders default when timedout', async () => {
    initialState.watch!.nextEpisodeTimeout = -1
    const wrapper = render(getComponent())
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('hides the button and calls next episode when clicking', async () => {
    const wrapper = render(getComponent())
    fireEvent.click(wrapper.getByTestId(NEXT_EPISODE_BUTTON.BUTTON))
    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        payload: false,
        type: 'watch/setNextEpisodeButton'
      })
    )
    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        type: 'watch/nextEpisode/pending'
      })
    )
  })
})
