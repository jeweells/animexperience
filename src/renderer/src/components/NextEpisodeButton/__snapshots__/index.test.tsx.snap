// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextEpisodeButton do not show when the episode is the last one 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-0:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <div
      style="opacity: 0; visibility: hidden;"
    >
      <div
        style="position: absolute; right: 16px; bottom: 16px; transform: translateY(-56px); transition: all 500ms ease;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
          data-testid="next-episode-button-nbutton"
          style="filter: invert(100%); overflow: hidden;"
          tabindex="0"
          type="button"
        >
          <span
            style="position: relative; z-index: 2;"
          >
            Siguiente episodio
          </span>
          <div
            style="position: absolute; z-index: 1; top: 0px; left: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.9); width: 0px; transition: none;"
          />
          <span
            class="MuiTouchRipple-root emotion-1"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`NextEpisodeButton renders default 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-0:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <div
      style="opacity: 0; visibility: hidden;"
    >
      <div
        style="position: absolute; right: 16px; bottom: 16px; transform: translateY(-56px); transition: all 500ms ease;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
          data-testid="next-episode-button-nbutton"
          style="filter: invert(100%); overflow: hidden;"
          tabindex="0"
          type="button"
        >
          <span
            style="position: relative; z-index: 2;"
          >
            Siguiente episodio
          </span>
          <div
            style="position: absolute; z-index: 1; top: 0px; left: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.9); width: 0px; transition: none;"
          />
          <span
            class="MuiTouchRipple-root emotion-1"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`NextEpisodeButton renders default when not shown 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #292d33;
  color: #e9ebf0;
  background-color: #292d33;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  outline: 0!important;
  white-space: nowrap;
  border: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 14px;
  line-height: 1.42857143;
  text-transform: unset;
  border-radius: 6px;
  height: 36px;
  color: #e9ebf0;
  text-transform: none;
  padding: 8px 12px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(41, 45, 51, 0.08);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-0:hover {
  outline: none;
  border: none;
  background-color: #3c3f43;
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body>
  <div>
    <div
      style="opacity: 0; visibility: hidden;"
    >
      <div
        style="position: absolute; right: 16px; bottom: 16px; transform: translateY(-56px); transition: all 500ms ease;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-0"
          data-testid="next-episode-button-nbutton"
          style="filter: invert(100%); overflow: hidden;"
          tabindex="0"
          type="button"
        >
          <span
            style="position: relative; z-index: 2;"
          >
            Siguiente episodio
          </span>
          <div
            style="position: absolute; z-index: 1; top: 0px; left: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.9); width: 0px; transition: none;"
          />
          <span
            class="MuiTouchRipple-root emotion-1"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`NextEpisodeButton renders default when timedout 1`] = `
<body>
  <div>
    <div
      style="opacity: 1; position: absolute; top: 16px; right: 16px; z-index: 2147483647; webkit-transition: opacity 400ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 400ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <style>
        
    @font-face {
        font-family: 'Quicksand';
        src: url('http://127.0.0.1:3000/mocked-font.ttf') format('truetype');
        font-weight: 300;
        font-style: normal;
    }

        

    button.raex-button-injected {
        font-family: 'Quicksand';
        color: #292d33;
        background: #e9ebf0;
        display: inline-block;
        margin-bottom: 0;
        font-weight: normal;
        text-align: center;
        vertical-align: middle;
        cursor: pointer;
        outline: 0 !important;
        white-space: nowrap;
        border: none;
        user-select: none;
        padding: 8px 12px;
        font-size: 14px;
        line-height: 1.42857143;
        border-radius: 6px;
        position: relative;
        transition: color 0.2s linear, background-color 0.3s linear;
        margin: 0;
        overflow: hidden;
        text-decoration: none;
    }
    button.raex-button-injected:hover, button.raex-button-injected:focus {
        color: #3c3f43;
        background-color: #e9ebf0;

    }

      </style>
      <button
        class="raex-button-injected"
        data-testid="next-episode-button-nbutton"
      >
        <span
          style="position: relative; z-index: 2;"
        >
          Siguiente episodio
        </span>
        <div
          style="position: absolute; z-index: 1; top: 0px; left: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.1); width: 0px; transition: none;"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`NextEpisodeButton renders when the episode is the last one 1`] = `
<body>
  <div>
    <div
      style="opacity: 0; visibility: hidden; position: absolute; top: 16px; right: 16px; z-index: 2147483647;"
    >
      <style>
        
    @font-face {
        font-family: 'Quicksand';
        src: url('http://127.0.0.1:3000/mocked-font.ttf') format('truetype');
        font-weight: 300;
        font-style: normal;
    }

        

    button.raex-button-injected {
        font-family: 'Quicksand';
        color: #292d33;
        background: #e9ebf0;
        display: inline-block;
        margin-bottom: 0;
        font-weight: normal;
        text-align: center;
        vertical-align: middle;
        cursor: pointer;
        outline: 0 !important;
        white-space: nowrap;
        border: none;
        user-select: none;
        padding: 8px 12px;
        font-size: 14px;
        line-height: 1.42857143;
        border-radius: 6px;
        position: relative;
        transition: color 0.2s linear, background-color 0.3s linear;
        margin: 0;
        overflow: hidden;
        text-decoration: none;
    }
    button.raex-button-injected:hover, button.raex-button-injected:focus {
        color: #3c3f43;
        background-color: #e9ebf0;

    }

      </style>
      <button
        class="raex-button-injected"
        data-testid="next-episode-button-nbutton"
      >
        <span
          style="position: relative; z-index: 2;"
        >
          Siguiente episodio
        </span>
        <div
          style="position: absolute; z-index: 1; top: 0px; left: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.1); width: 0px; transition: all 5s;"
        />
      </button>
    </div>
  </div>
</body>
`;
